crates/shared/src/current_block.rs

```diff
@@ -298,4 +302,15 @@ mod tests {
         assert_eq!(blocks.last().unwrap().0, 5);
         assert_eq!(blocks.first().unwrap().0, 0);
     }
+
+    #[test]
+    fn range_contains() {
+        let range = RangeInclusive::try_new(42, 1337).unwrap();
```

Comment:
Why not use https://doc.rust-lang.org/std/ops/struct.RangeInclusive.html?

Reply:
I don't know, I see the event parsing was handled by @sunce86 so he might have more details. The only practical difference I see is that the custom implementation cannot be created if `start >= end`.